FWCore/Catalog/interface/FileLocator.h

```diff
@@ -1,22 +1,31 @@
 #ifndef FWCore_Catalog_FileLocator_h
 #define FWCore_Catalog_FileLocator_h
 
+#include "FWCore/Catalog/interface/SiteLocalConfig.h"
 #include <string>
 #include <list>
 #include <map>
 #include <utility>
 #include <regex>
 #include "tinyxml2.h"
+#include <boost/property_tree/ptree.hpp>
 
 namespace edm {
 
   class FileLocator {
   public:
+    explicit FileLocator(
+        edm::CatalogAttributes const& catAttr,
+        unsigned iCatalog = 0,
+        std::string storageDescriptionPath = std::
+            string());  //storageDescriptionPath is used to override path provided by SiteLocalConfig. This is used in FileLocator_t.cpp tests
     explicit FileLocator(std::string const& catUrl, unsigned iCatalog = 0);
+
     ~FileLocator();
 
-    std::string pfn(std::string const& ilfn) const;
-    std::string lfn(std::string const& ipfn) const;
+    std::string pfn(std::string const& ilfn, edm::CatalogType catType) const;
+
+    std::string prefix() const;
```

Comment:
Is this function defined or used anywhere? I did not find any references to it in this PR.